File: Backend/app/routers/retailer.py
"""
Dashboard B2B revendeur (Feature 2) — vue agrégée par zone géographique pour
le réseau de distribution Michelin : profil de pratique dominant et pneus
proches de la fin de vie. Pas d'authentification dédiée revendeur (décision
produit : nouvelle page interne de démonstration, pas de vrai compte
revendeur dans ce projet).
"""
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.database import get_db
from app import models, schemas, retailer

router = APIRouter(prefix="/retailer", tags=["retailer"])


@router.get("/dashboard", response_model=schemas.RetailerDashboardOut)
def get_dashboard(db: Session = Depends(get_db)):
    profiles = db.query(models.DemoRiderProfile).all()
    catalog_by_id = {c.id: c for c in db.query(models.TireCatalog).all()}
    zones = retailer.aggregate_by_zone(profiles, catalog_by_id)
    return schemas.RetailerDashboardOut(
        zones=[schemas.RetailerZoneOut(**z) for z in zones],
        total_riders=len(profiles),
        weeks_horizon=retailer.WEEKS_HORIZON,
    )

File: Frontend/app/lib/api/retailer.ts
import type { RetailerDashboardOut } from './types';
import api from './client';

export const retailerApi = {
	/** GET /retailer/dashboard — vue agrégée par zone (profils dominants + pneus en fin de vie) */
	getDashboard: (): Promise<RetailerDashboardOut> =>
		api.get<RetailerDashboardOut>('/retailer/dashboard').then((r) => r.data),
};
